### PR TITLE
Adds getClosestMatchingDisplayMode

### LINKED ISSUE #471: Add getClosestMatchingDisplayMode
Wraps the `SDL_GetClosestDisplayMode` as below. It removes the need to declare the implementation-specific `extern SDL_Window* underlyingWindow;` in game code.
```
NAS2D::DisplayDesc getClosestMatchingDisplayMode(const DisplayDesc& preferred)
{
    const auto display_index = SDL_GetWindowDisplayIndex(underlyingWindow);
    SDL_DisplayMode desired{preferred.width, preferred.height}; //Filter by resolution only. Default other members to desktop.
    SDL_DisplayMode closest{};
    if(SDL_GetClosestDisplayMode(display_index, &desired, &closest))
    {
        return NAS2D::DisplayDesc{closest.w, closest.h, closest.refreshRate};
    }
    const auto err_str = std::string{__FUNCTION__} + " no closest matching display mode available."
    throw std::runtime_error(err_str);
}
```

### PR DESCRIPTION
**Pull request by cugone**
Closes #471

### CODE REVIEW

File: src/Renderer/RendererOpenGL.cpp [hunk 1]
```diff
@@ -764,6 +764,21 @@ std::vector<NAS2D::DisplayDesc> NAS2D::RendererOpenGL::getDisplayModes() const
...
+	preferred.w = preferredDisplayDesc.width;
+	preferred.h = preferredDisplayDesc.height;
+	preferred.refresh_rate = preferredDisplayDesc.refreshHz;
+
+	SDL_DisplayMode closest{};
+	if (SDL_GetClosestDisplayMode(display_index, &preferred, &closest))
+	{
+		return {closest.w, closest.h, closest.refresh_rate};
+	}
+	return {};
```
Comment (DanRStevens) [inline src/Renderer/RendererOpenGL.cpp]: Might be better to `throw` rather than silently return an unusable result.

Comment (cugone) [reply src/Renderer/RendererOpenGL.cpp]: I was following what the underlying implementations do: SDL just returns a `nullptr` and DirectX returns an `HRESULT` denoting an error code. Neither throw.

Could document that if one isn't found a null description (all zeros) is returned.

Comment (DanRStevens) [reply src/Renderer/RendererOpenGL.cpp]: Neither of those can really throw. Exception handling with `throw` is C++ specific. Those libraries are fairly language agnostic.

It's up to the wrapper code to translate library error signalling to language specific error signalling.

Comment (cugone) [reply src/Renderer/RendererOpenGL.cpp]: Makes sense.

**Pull request merged**